rotkehlchen/globaldb/cache.py

```diff
@@ -1,77 +1,120 @@
+BASE_POOL_TOKENS_KEY_LENGTH = len(compute_cache_key([CacheType.CURVE_POOL_TOKENS, ZERO_ADDRESS])[0])  # noqa: E501
 
 
-def globaldb_set_general_cache_values_at_ts(
+def globaldb_set_cache_values_at_ts(
         write_cursor: DBCursor,
-        key_parts: Iterable[Union[str, GeneralCacheType]],
+        key_parts: Iterable[Union[str, CacheType]],
         values: Iterable[str],
         timestamp: Timestamp,
 ) -> None:
     """Function to update cache in globaldb. Inserts all values paired with the cache key.
     If any entry exists, overwrites it."""
-    cache_key = compute_cache_key(key_parts)
+    cache_key, cache_table = compute_cache_key(key_parts)
     tuples = [(cache_key, value, timestamp) for value in values]
     write_cursor.executemany(
-        'INSERT OR REPLACE INTO general_cache '
+        f'INSERT OR REPLACE INTO {cache_table} '
         '(key, value, last_queried_ts) VALUES (?, ?, ?)',
         tuples,
     )
 
 
-def globaldb_set_general_cache_values(
+def globaldb_set_cache_values(
         write_cursor: DBCursor,
-        key_parts: Iterable[Union[str, GeneralCacheType]],
+        key_parts: Iterable[Union[str, CacheType]],
         values: Iterable[str],
 ) -> None:
     """Function to update cache in globaldb. Inserts all values paired with the cache key.
     If any entry exists, overwrites it. The timestamp is always the current time."""
     timestamp = ts_now()
-    globaldb_set_general_cache_values_at_ts(
+    globaldb_set_cache_values_at_ts(
         write_cursor=write_cursor,
         key_parts=key_parts,
         values=values,
         timestamp=timestamp,
     )
 
 
-def globaldb_get_general_cache_values(
+@overload
+def globaldb_get_cache_values(
         cursor: DBCursor,
-        key_parts: Iterable[Union[str, GeneralCacheType]],
+        key_parts: Union[tuple[UniqueCacheType, str], tuple[UniqueCacheType]],
+) -> Optional[str]:
+    ...
+
+
+@overload
+def globaldb_get_cache_values(
+        cursor: DBCursor,
+        key_parts: Union[tuple[GeneralCacheType, str], tuple[GeneralCacheType]],
 ) -> list[str]:
-    """Function to read globaldb cache. Returns all the values that are paired with the key."""
-    cache_key = compute_cache_key(key_parts)
-    cursor.execute('SELECT value FROM general_cache WHERE key=?', (cache_key,))
-    return [entry[0] for entry in cursor]
+    ...
+
+
+def globaldb_get_cache_values(
+        cursor: DBCursor,
+        key_parts: Iterable[Union[str, CacheType]],
+) -> Union[Optional[str], list[str]]:
+    """Function to read globaldb cache. It reads either from the general cache or the unique cache
+    table. If it reads from the general cache, it returns all the values that are paired
+    with the given key."""
+    cache_key, cache_table = compute_cache_key(key_parts)
+    cursor.execute(f'SELECT value FROM {cache_table} WHERE key=?', (cache_key,))
+    results = [entry[0] for entry in cursor]
+
+    if next(iter(key_parts)) in UNIQUE_CACHE_KEYS:
```

Comment:
Why do this? I don't really understand why you need to do `next(iter(key_parts))` here? The `next(iter(key_parts)) in UNIQUE_CACHE_KEYS` check already happened in `compute_cache_key` right? 
```suggestion
    if cache_table == UNIQUE_CACHE_TABLE_NAME:
```